packages/block-editor/src/components/rich-text/use-format-types.js

```diff
@@ -71,16 +71,18 @@ export function useFormatTypes( {
 				return false;
 			}
 
+			// The "Footnote" format requires special handling due to its nested interactive content tag.
 			if (
 				withoutInteractiveFormatting &&
-				interactiveContentTags.has( tagName )
+				( name === 'core/footnote' ||
+					interactiveContentTags.has( tagName ) )
```

Comment:
Maybe we should add a property to the format registration object to let formats declare themselves as interactive (in addition to checking the tag of course).

Reply:
Updated in [638bac2](https://github.com/WordPress/gutenberg/pull/53474/commits/638bac255a534e915e9b65ae4d7433e24c607765).

Sidenote: It would be nice to document some "recent" additions to format-type settings like `object`, `contentEditable` and `attributes`. I know @fabiankaegy has requested this before - #40051.